test/AutoRest.TestServer.Tests/TypeSchemaMappingTest.cs

```diff
@@ -86,13 +88,53 @@ public void ObjectsAreMappedToSchemas()
             Assert.AreEqual(1, modelType.GetConstructors(BindingFlags.Instance | BindingFlags.NonPublic).Length);
             Assert.AreEqual(1, modelType.GetConstructors(BindingFlags.Instance | BindingFlags.Public).Length);
 
-            PropertyInfo firstProperty = TypeAsserts.HasProperty(modelType, "CustomizedETagProperty", BindingFlags.Instance | BindingFlags.NonPublic);
+            PropertyInfo firstProperty = TypeAsserts.HasProperty(modelType, "CustomizedETagProperty", BindingFlags.Instance | BindingFlags.Public);
             Assert.AreEqual(typeof(ETag), firstProperty.PropertyType);
 
-            PropertyInfo secondProperty = TypeAsserts.HasProperty(modelType, "CustomizedCreatedAtProperty", BindingFlags.Instance | BindingFlags.NonPublic);
+            PropertyInfo secondProperty = TypeAsserts.HasProperty(modelType, "CustomizedCreatedAtProperty", BindingFlags.Instance | BindingFlags.Public);
             Assert.AreEqual(typeof(DateTime), secondProperty.PropertyType);
         }
 
+        [Test]
+        public void ObjectTypePropertiesSerializedAsValues()
+        {
+            DateTime date = DateTime.UtcNow;
+            var inputModel = new RenamedThirdModel(new ETag("Id"), date);
+
+            JsonAsserts.AssertSerialization(
+                @"{""ETag"":""Id"",""CreatedAt"":" + JsonSerializer.Serialize(date) + "}",
+                inputModel);
+        }
+
+        [Test]
+        public void ObjectTypePropertiesDeserializedAsValues()
+        {
+            DateTime date = DateTime.UtcNow;
+            RenamedThirdModel model = RenamedThirdModel.DeserializeRenamedThirdModel(JsonDocument.Parse("{\"ETag\":\"ETagValue\", \"CreatedAt\":" + JsonSerializer.Serialize(date) + "}").RootElement);
+
+            Assert.AreEqual("ETagValue", model.CustomizedETagProperty.ToString());
+            Assert.AreEqual(date, model.CustomizedCreatedAtProperty);
+        }
+
+        [Test]
+        public void ObjectTypePropertiesSerializedAsNull()
+        {
+            var inputModel = new RenamedThirdModel();
```

Comment:
Let's put some value into ETag property

Reply:
[ObjectTypePropertiesSerializedAsValues](https://github.com/Azure/autorest.csharp/blob/3a7350da027134ff2a96431e4a27738e65b60213/test/AutoRest.TestServer.Tests/TypeSchemaMappingTest.cs#L99) test covers this.